### ISSUE TITLE
Instruction on how to compile to generate UEFI binary

### ISSUE THREAD

**Issue by sandeep-gh** [pftf/RPi4#261]
The need to compile comes from disabling by 3GB memory limit when netbooting a Rpi4. I am surprised that there is no clear instruction on how to go about compiling the source code to generate the UEFI binary. FWIW, i had to go through these hoops to successfully compile. 

```      
      mkdir RPi4
      export WORKSPACE=$(pwd)/RPi4
      git clone http://github.com/pftf/RPi4.git
      cd RPi4/
      sudo apt install python3-pip
      python3 -m venv venv
      . ./venv/bin/activate
      git submodule update --init
      cd edk2
      git submodule update --init
      git submodule update --init --recursive
      pip3 install -r pip-requirements.txt
      make -C BaseTools
      mkdir -p /home/adming/RPi4/Conf

      export EDK_TOOLS_PATH=$PWD/BaseTools 
      source edksetup.sh 

      cd ..
      ln -s  edk2-platforms/Platform/ .
      cd -
      cp -r ~/RPi4/edk2/MdePkg/ .
      cd Platform/RaspberryPi/
      cd RPi4/
      cp -r ~/RPi4/edk2/NetworkPkg/ .
      cp -r ~/RPi4/edk2/ArmPlatformPkg/ .
      cd ~/RPi4/
      cp -r ~/RPi4/edk2/ArmPlatformPkg/ .
      cp -r ~/RPi4/edk2/MdeModulePkg/ .
      cp -r ~/RPi4/edk2/MdePkg/ .
      cp -r ~/RPi4/edk2/ArmPkg/ .
      cp -r ~/RPi4/edk2/EmbeddedPkg/ .
      cp -r ~/RPi4/edk2/CryptoPkg/ .
      cp -r edk2-platforms/Silicon/  .
      cp -r ~/RPi4/edk2/ShellPkg/ .
      cp -r ~/RPi4/edk2/NetworkPkg/ .
      cp -r ~/RPi4/edk2/FatPkg/ .
      cd /home/adming/RPi4/Platform/RaspberryPi/Drivers
      cd /home/adming/RPi4/Platform/RaspberryPi/Drivers
      cp -r ~/RPi4/edk2-non-osi/Platform/RaspberryPi/Drivers/LogoDxe/ .
      pip install --upgrade edk2-basetools
      sudo apt-get install acpica-tools

      sudo apt-get install uuid-dev
      make -C /home/adming/RPi4/edk2/BaseTools/Source/C
      cp -r ~/RPi4/edk2-non-osi/Platform/RaspberryPi/RPi4/TrustedFirmware .
      build -a AARCH64 -t GCC5 -p Platform/RaspberryPi/RPi4/RPi4.dsc --pcd gRaspberryPiTokenSpaceGuid.PcdRamLimitTo3GB=0 
```

I am sure it should be simpler then this. But unable to find any doc that tell comprehensively how to build the code. 

Thanks
-S

**Comment by pbatard** [comment]
I think you misunderstand the goal of this repository.

It's not a repository that aims at helping you compile a Raspberry Pi firmware on your own. If you want to do that, you should follow what the documentation from https://github.com/tianocore/edk2-platforms/ and https://github.com/tianocore/edk2-platforms/tree/master/Platform/RaspberryPi/RPi4 (which simply directs you to the former).

Instead, it's a repository that aims at providing ready to use binaries that you can download.

In other words, this does **not** supersede https://github.com/tianocore/edk2-platforms/tree/master/Platform/RaspberryPi/RPi4 or https://github.com/tianocore/edk2-platforms/tree/master/Platform/RaspberryPi/RPi3, which is the **only** official source for compiling a Raspberry Pi firmware and which is what you should use as your base, rather than this repo.
